src/sentry/static/sentry/app/views/usageStats/usageChart/utils.tsx

```diff
@@ -1,26 +1,56 @@
 import moment from 'moment';
 
-import {DataCategory} from 'app/types';
+import {parseStatsPeriod} from 'app/components/organizations/globalSelectionHeader/getParams';
+import {DataCategory, IntervalPeriod} from 'app/types';
+import {intervalToMilliseconds} from 'app/utils/dates';
 
 import {formatUsageWithUnits} from '../utils';
 
-export function getDateFromMoment(m: moment.Moment) {
-  return m.format('MMM D');
+/**
+ * Avoid changing "MMM D" format as X-axis labels on UsageChart are naively
+ * truncated by date.slice(0, 6). This avoids "..." when truncating by ECharts.
+ */
+export const FORMAT_DATETIME_HOURLY = 'MMM D LT';
+export const FORMAT_DATETIME_DAILY = 'MMM D';
+
+/**
+ * Used to generate X-axis data points and labels for UsageChart
+ * Ensure that this method is idempotent and doesn't change the moment object
+ * that is passed in
+ */
+export function getDateFromMoment(m: moment.Moment, interval: IntervalPeriod = '1d') {
+  const days = intervalToMilliseconds(interval) / (1000 * 60 * 60 * 24);
```

Comment:
Should we make `(1000 * 60 * 60 * 24)` into a module level constant?

Reply:
This is fixed in https://github.com/getsentry/sentry/pull/25211/ (https://github.com/getsentry/sentry/pull/25211/commits/44c72a6fd04a2a07645ceccc83d34a9b5fee8631)